generateGraph.ts

```diff
@@ -1,157 +1,456 @@
+            );
+
+            
+            //------------------
+            // Draw sensor data:
+            //------------------
+
+            // Draw the data from each sensor, as a separate coloured line: sensors may have variable quantities of data:
+            for (let sensor = 0; sensor < this.numberOfSensors; sensor++) {
+                for (let row = 0; row < this.processedReadings[sensor].length; row+=2) {
+                    const xOffset = ((row / 2) * this.xCoordinateScalar);
+                    screen.drawLine(
+                        this.windowLeftBuffer + xOffset,
+                        this.processedReadings[sensor][row],
+                        this.windowLeftBuffer + xOffset + this.xCoordinateScalar,
+                        this.processedReadings[sensor][row + this.numberOfSensors],
+                        SENSOR_COLORS[sensor % SENSOR_COLORS.length]
+                    );
+                }
+            }
+
+            
+            //---------------
+            // Sensor blocks:
+            //---------------
+
+            let y = this.windowHeight - 2 + (2 * this.yScrollOffset)
+            for (let i = 0; i < this.numberOfSensors; i++) {
+                // Black edges:
+                screen.fillRect(
+                    5,
+                    y,
+                    142,
+                    47,
+                    15
+                )
+
+                // Sensor is disabled:
+                let blockColor: number = SENSOR_COLORS[(i % this.numberOfSensors) % SENSOR_COLORS.length]
+                let textColor: number = 15; // black
+                if (!this.drawSensorStates[i]) {
+                    blockColor = 15; // black
+                    textColor = 1;   // white
+                }
+
+                // Coloured block:
+                screen.fillRect(
+                    7,
+                    y,
+                    145,
+                    45,
+                    blockColor
+                )
+
+                // Blue outline for selected sensor:
+                if (this.uiState == UI_STATE.SENSOR_SELECTION && i == this.currentlySelectedSensorIndex) {
+                    // Blue edges:
+                    for (let thickness = 0; thickness < 3; thickness++) {
+                        screen.drawRect(
+                            7 - thickness,
+                            y - thickness,
+                            145 + thickness,
+                            45 + thickness,
+                            6
+                        )
+                    }
+                }
+
+                //--------------------
+                // Sensor information:
+                //--------------------
+
+                screen.print(
+                    this.sensorNames[i],
+                    12,
+                    y + 2,
+                    textColor
+                )
+
+                screen.print(
+                    "Minimum: " + this.sensorMinsAndMaxs[i][0],

```

Comment:
named constants for the `sensorMinsAndMaxs` second indexer would be helpful for readability.
```suggestion
                    "Minimum: " + this.sensorMinsAndMaxs[i][MIN],

```